src/mesh/NodeDB.cpp

```diff
@@ -475,6 +475,15 @@ bool saveProto(const char *filename, size_t protoSize, size_t objSize, const pb_
             DEBUG_MSG("Error: can't rename new pref file\n");
     } else {
         DEBUG_MSG("Can't write prefs\n");
+#ifdef ARCH_NRF52
+        static uint8_t failedCounter = 0;
+        failedCounter++;
+        if(failedCounter >= 2){
+            FSCom.format();
+            //After formatting, the device needs to be restarted
+            nodeDB.resetRadioConfig();
+        }
```

Comment:
resetRadioConfig will only reboot the device if called as resetRadioConfig(true) inducing a factory reset. You can just use `rebootAtMsec = millis() + (5 * 1000);` instead since on a freshly formatted filesystem the default config will be written on next boot anyway.

Reply:
Changes have been made